### DISCUSSION TITLE
[Script request] CodeProject.AI Server

### DISCUSSION THREAD

**Discussion by brnu26** [community-scripts/ProxmoxVE#828] [Request script]
### Application Name

CodeProject.AI Server

### Website

https://github.com/codeproject/CodeProject.AI-Server

### Description

A standalone, self-hosted, fast, free and Open Source Artificial Intelligence microserver for any platform, any language. It can be installed locally, required no off-device or out of network data transfer, and is easy to use. It can run any AI module your imagination and patience can create.

Current modules include:

    Generative AI: LLMs for text generation, Text-to-image, and multi-modal LLMs (eg "tell me what's in this picture")
    Object Detection in images, including using custom models
    Faces detection and recognition images
    Scene recognition represented in an image
    Remove a background from an image
    Blur a background from an image
    Enhance the resolution of an image
    Pull out the most important sentences in text to generate a text summary
    Prove sentiment analysis on text
    Sound Classification
    
This can be used alongside the already existing AgentDVR script to add object detection.

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.
- [X] This is not a game-related request.

**Comment by Boss-man** [comment]
Someone did a writeup on how they did it here, https://forum.proxmox.com/threads/proxmox-debian-12-lxc-codeproject-ai-coral-usb-tpu-my-working-setup.148086

For 2.6.5 version and with a coral tpu, so change some urls for the 2.9.5 version and omit the coral stuff, you should be running.

**Reply by brnu26** [reply]
Thanks for this! I condensed this to what I did for using only YOLOv5 .NET object detection, still would like to see a script added.

1. Use Debian LXC Helper Script:

`bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/debian.sh)"`

4 cpu, 5gb disk, 2gb ram

2. Install .NET SDK and required dependencies:

```
wget https://packages.microsoft.com/config/debian/12/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
dpkg -i packages-microsoft-prod.deb
rm packages-microsoft-prod.deb
apt update
apt install dotnet-sdk-9.0 psmisc jq
```
3. Install CodeProject.AI:

```
wget https://codeproject-ai-bunny.b-cdn.net/server/installers/linux/codeproject.ai-server_2.9.5_Ubuntu_x64.zip
unzip codeproject.ai-server_2.9.5_Ubuntu_x64.zip
rm codeproject.ai-server_2.9.5_Ubuntu_x64.zip
dpkg -i codeproject.ai-server_2.9.5_Ubuntu_x64.deb
rm codeproject.ai-server_2.9.5_Ubuntu_x64.deb
```
4. Start CodeProject.AI:

`pushd "/usr/bin/codeproject.ai-server-2.9.5/" && bash start.sh && popd`

Ctrl+C at "Infor Module ObjectDetectionYOLOv5-6.2 started successfully."

5. Enable CodeProject.AI Service:

`systemctl enable codeproject.ai-server`

6. Reboot and go to http://[container-ip]:32168, start Object Detection (YOLOv5 .NET), uninstall Object Detection (YOLOv5 6.2) and Face Processing
